src/huggingface_hub/repocard_types.py

```diff
@@ -72,9 +104,9 @@ class SingleResultDataset:
 
 @dataclass
 class SingleResult:
-    metrics: "List[SingleMetric]"
     task: "SingleResultTask"
     dataset: "Optional[SingleResultDataset]"
+    metrics: "List[SingleMetric]"
```

Comment:
I switched the order here as it's more natural to read the metadata as task > dataset > metrics

Reply:
(this is – or used to be – generated code)

Reply:
(see first line of file)

Reply:
thanks for the pointer - do you happen to know where the original JSON TypeDef schema is where we can generate this from?

Reply:
inside moon-landing, but we may be in the middle of dropping it and switching to a differnt system, so no big deal (cc @coyotte508)

Reply:
relevant internal PR: https://github.com/huggingface/moon-landing/pull/2960

Reply:
OK, we'll open a separate PR on `huggingface_hub` that just updates these Python classes with the new fields (since this PR won't get merged anyway)

cc @TristanThrush